internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -990,3 +997,14 @@ func FindStatusCondition(conditions []hivev1.ClusterInstallCondition, conditionT
 
 	return nil
 }
+
+// ensureOwnerRef sets the owner reference of ClusterDeployment on AgentClusterInstall
+func (r *ClusterDeploymentsReconciler) ensureOwnerRef(ctx context.Context, cd *hivev1.ClusterDeployment, ci *hiveext.AgentClusterInstall) error {
+
+	if err := controllerutil.SetOwnerReference(cd, ci, r.Scheme); err != nil {
```

Comment:
I'm playing around with this to set the finalized on `agentClusterInstall`.
based on what I see, and the [documentation](https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/#foreground-cascading-deletion) about garbage-collection: Shouldn't we use `blockOwnerDeletion=true` here?

I see `SetOwnerReference` does not set any value for `blockOwnerDeletion`
https://github.com/kubernetes-sigs/controller-runtime/blob/ce2f0c92d77e05609fab003663f2b5bb9c1990d7/pkg/controller/controllerutil/controllerutil.go#L111-L116

And the default to `false`
https://github.com/kubernetes/apimachinery/blob/8daf28983e6ecf28bd8271925ee135c1179ad13a/pkg/apis/meta/v1/types.go#L325-L329

from what I

Reply:
After some additional discussion and experimenting with `blockOwnerDeletion=true`, it doesn't seem like it does what we thought it will. 
It will not delete `ACI` before `CD` without `propagationPolicy":"Foreground"`, which is a Client attribute.

So please disregard this comment.